src/target/riscv/riscv-013.c

```diff
@@ -4486,27 +4480,21 @@ static int write_memory_bus_v1(struct target *target, target_addr_t address,
 
 			uint32_t sbvalue[4] = { 0 };
 			if (size > 12) {
-				sbvalue[3] = ((uint32_t)p[12]) |
-						(((uint32_t)p[13]) << 8) |
-						(((uint32_t)p[14]) << 16) |
-						(((uint32_t)p[15]) << 24);
+				sbvalue[3] = buf_get_u32(&p[12],
+						/* first = */ 0, /* bit_num = */ 32);
 				riscv_batch_add_dm_write(batch, DM_SBDATA3, sbvalue[3], false,
 						RISCV_DELAY_BASE);
 			}
 
 			if (size > 8) {
-				sbvalue[2] = ((uint32_t)p[8]) |
-						(((uint32_t)p[9]) << 8) |
-						(((uint32_t)p[10]) << 16) |
-						(((uint32_t)p[11]) << 24);
+				sbvalue[2] = buf_get_u32(&p[8],
+						/* first = */ 0, /* bit_num = */ 32);
 				riscv_batch_add_dm_write(batch, DM_SBDATA2, sbvalue[2], false,
 						RISCV_DELAY_BASE);
 			}
 			if (size > 4) {
-				sbvalue[1] = ((uint32_t)p[4]) |
-						(((uint32_t)p[5]) << 8) |
-						(((uint32_t)p[6]) << 16) |
-						(((uint32_t)p[7]) << 24);
+				sbvalue[1] = buf_get_u32(&p[4],
```

Comment:
```suggestion
				sbvalue[1] = buf_get_u32(&(p[4]),
```

Reply:
Looks like checkpatch doesn't allow this

Reply:
Ok, no problem, thanks for trying it.